webhook_server/utils/helpers.py

```
@@ -492,6 +492,23 @@ def get_api_with_highest_rate_limit(config: Config, repository_name: str = "") -
     apis_and_tokens = get_apis_and_tokes_from_config(config=config)
     logger.debug(f"Checking {len(apis_and_tokens)} API(s) for highest rate limit")
 
+    # Short-circuit: single token doesn't need rate limit comparison
+    if len(apis_and_tokens) == 1:
+        _api, _token = apis_and_tokens[0]
+        if _api.rate_limiting[-1] == 60:
+            raise NoApiTokenError("Single configured token has rate limit 60 (indicates invalid token)")
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)